internal/hardware/validator_test.go

```diff
@@ -349,7 +350,23 @@ var _ = Describe("Disk eligibility", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(eligible).To(BeEmpty())
 
-		By("Check infra env FC is only eligible for s390x")
+		By("Check infra env FC is eligible for s390x")
+		eligible, err = hwvalidator.DiskIsEligible(ctx, &testDisk, infraEnv, nil, &host, inventory)
+
+		Expect(err).ToNot(HaveOccurred())
+		Expect(eligible).To(BeEmpty())
+	})
+
+	It("Check if FC is eligible for x86_64", func() {
+		testDisk.DriveType = models.DriveTypeFC
+		inventory.CPU.Architecture = models.ClusterCPUArchitectureX8664
+
+		eligible, err := hwvalidator.DiskIsEligible(ctx, &testDisk, infraEnv, &cluster, &host, inventory)
+
+		Expect(err).ToNot(HaveOccurred())
+		Expect(eligible).To(BeEmpty())
```

Comment:
I think it may be better to rename `eligible` to `noteligiblereasons` as per the return from the function under test.

The way this tests reads otherwise is a little misleading.

I think it should be clear that we are looking for no error + no reasons for ineligibility.

Reply:
done